include/envoy/server/worker.h

```diff
@@ -100,11 +100,12 @@ class WorkerFactory {
   virtual ~WorkerFactory() = default;
 
   /**
+   * @param id supplies the id of the worker, in the range of [0, concurrency).
    * @param overload_manager supplies the server's overload manager.
    * @param worker_name supplies the name of the worker, used for per-worker stats.
    * @return WorkerPtr a new worker.
    */
-  virtual WorkerPtr createWorker(OverloadManager& overload_manager,
+  virtual WorkerPtr createWorker(uint32_t id, OverloadManager& overload_manager,
```

Comment:
nit: if we stick with passing the id around, I would prefer index vs. id since that is more what it is, though it would be nice to hide this in the implementation somehow? Is this possible?